### PR TITLE
feat: implement list creation from insights contributors page

### LINKED ISSUE #1156: feat: Implement ability to create list on contributors tab
This requires the following before be started

https://github.com/open-sauced/insights/issues/1153
https://github.com/open-sauced/insights/issues/1154

Screen Shot 2023-05-08 at 10 07 28 PM

## Design 

[Figma](https://www.figma.com/file/768ifqOGWrsRN29Igkc4ip/User-Flows?type=design&node-id=205-1661&t=eleETzJqGQfeLI12-0)

### PR DESCRIPTION
**Pull request by OgDev-01**
## Description
This PR Adds the ability to add contributors to new or existing lists from the contributors table on insights and explore pages

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
Closes #1156

## Mobile & Desktop Screenshots and Recordings

https://github.com/open-sauced/app/assets/62995161/7af563e0-354f-4822-a233-4d06d3645e92

https://github.com/open-sauced/app/assets/62995161/ae9faefc-500e-4f2d-86c8-127b7f529004

### Mobile
image

image

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by isabensusan** [comment]
loving the select component! I would love to use the same component (the multiselect) and the same interaction in both the Contributor Table and on User profiles.

We want to use the multi select on both places (user profiles and on contributors table) to add a user to a list. I would do one small change to the component and add 2 CTAs, one to apply the "add to list" action and one to create a new list. This is because the interaction of clicking away from the component to add a user to a list feels a bit off, and having a cta to apply the change would be better. In this case, if the user clicks outside the component, it wouldn't add the user to the list, it would just close the select component.

image

>

**Comment by OgDev-01** [comment]
> loving the select component! I would love to use the same component (the multiselect) and the same interaction in both the Contributor Table and on User profiles.
> 
> We want to use the multi select on both places (user profiles and on contributors table) to add a user to a list. I would do one small change to the component and add 2 CTAs, one to apply the "add to list" action and one to create a new list. This is because the interaction of clicking away from the component to add a user to a list feels a bit off, and having a cta to apply the change would be better. In this case, if the user clicks outside the component, it wouldn't add the user to the list, it would just close the select component.
> 
> image
> >

Brilliant 👍. I'll implement this changes shortly 🍕

**Comment by OgDev-01** [comment]
Will mark this as ready for review once we land https://github.com/open-sauced/api/pull/401

**Comment by nickytonline** [comment]
@OgDev-01, as @bdougie mentioned in #1978, there significant overlap with the work in #2016. The only difference is selecting a list whereas in #2016, the list is already set.

All we need to do is add a list drop down in the header to set the list before adding the contributors.

Let's pause on this draft PR and we can sync up about this.

**Review by brandonroberts — APPROVED** [review]

**Review by bdougie — APPROVED** [review]
Works as expected

**Comment by brandonroberts** [comment]
@nickytonline will you take a second look and see if this is ready to go?

File: components/organisms/Contributors/contributors.tsx [hunk 1]
```diff
@@ -51,6 +79,103 @@ const Contributors = ({ repositories }: ContributorProps): JSX.Element => {
...
+        .then((res) => {
+          toast({
+            description: `Successfully added ${selectedContributors.length} contributors to ${selectedListIds.length} lists!`,
+            variant: "success",
+          });
+        })
+        .catch((res) => {
+          toast({
+            description: `
+            An error occurred while adding contributors to lists. Please try again later.
```
Comment (nickytonline) [inline components/organisms/Contributors/contributors.tsx:110]: wording (non-blocking): Saying try again is probably enough. No need for the later.

```suggestion
            description: `
            An error occurred while adding contributors to lists. Please try again.
```

**Review by nickytonline — APPROVED** [review]
This is awesome. Great work @OgDev-01! 🚀

**Pull request merged**